File: services/ms-auth/src/main/java/airportsystem/msauth/config/RabbitMQConfig.java
package airportsystem.msauth.config;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.json.JsonMapper;
import org.springframework.amqp.core.Queue;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class RabbitMQConfig {

    @Bean
    public Queue usuarioLoginQueue() { return new Queue("auth.login", false); }

    @Bean
    public Queue usuarioCriarQueue() { return new Queue("auth.criar", false); }

    @Bean
    public Queue usuarioAlterarQueue() { return new Queue("auth.alterar", false); }

    @Bean
    public Queue usuarioInativarQueue() { return new Queue("auth.inativar", false); }

    @Bean
    public Queue retornoQueue() {
        return new Queue("retorno", false);
    }


    @Bean
    public ObjectMapper objectMapper() {
        return JsonMapper.builder().findAndAddModules().build();
    }
}

File: services/ms-auth/airport-system.ms-auth/src/main/kotlin/com/tads/airport_system/ms_auth/config/RabbitMQConfig.kt
package com.tads.airport_system.ms_auth.config

import org.springframework.amqp.core.Queue
import org.springframework.amqp.rabbit.connection.ConnectionFactory
import org.springframework.amqp.rabbit.core.RabbitTemplate
import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class RabbitMQConfig {
    
    @Bean
    fun authServiceQueue(): Queue {
        return Queue("auth-service-queue", true)
    }

    @Bean
    fun authServiceExchange(): DirectExchange{
        return DirectExchange("auth")
    }

    @Bean
    fun binding(
        authServiceQueue: DirectExchange,
        authServiceExchange: Queue
    ):Binding{
        return BindingBuilder.bind(authServiceQueue).to(authServiceExchange).with("auth")
    }

    @Bean
    fun jsonMessageConverter(): Jackson2JsonMessageConverter {
        return Jackson2JsonMessageConverter()
    }
    
    @Bean
    fun rabbitTemplate(connectionFactory: ConnectionFactory): RabbitTemplate {
        val rabbitTemplate = RabbitTemplate(connectionFactory)
        rabbitTemplate.messageConverter = jsonMessageConverter()
        return rabbitTemplate
    }
} 